src/components/Input/Input.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from '../../packages/bem';
+import styles from './Input.scss';
+import { CONTEXTS, INPUT_TYPES, SIZES } from '../../constants';
+
+const { block } = bem({
+    name: 'Input',
+    classnames: styles,
+    propsToMods: ['context', 'size']
+});
+
+const Input = props => {
+    const { children, context, disabled, onChange, size, type, ...rest } = props;
+
+    return (
+        <input {...rest} {...block(props)} type={type} onChange={onChange} disabled={disabled} />
+    );
+};
+
+Input.propTypes = {
+    /** The input field context (e.g. brand, primary, bad, good etc. - defaults to brand) */
+    context: PropTypes.oneOf(CONTEXTS),
+    /** Should the input field be disabled or not */
+    disabled: PropTypes.bool,
+    /** Callback function on change */
+    onChange: PropTypes.func,
```

Comment:
Trying to come up with some solid idea about which event handlers should be listed in docs, and which can be passed silently by `{ ... rest }`. To list them all is stupid, to list only some of them can be confusing (like we support only that one), list none of them... don't know, but it's consistent.
@carlobernardini @ribeiroguilherme @eszthoff what do you think?

Reply:
I think you're right. But if we don't define any of them, none of them will have any validation. Defining all of them isn't a good idea and would take a lot of work.
I can't think of a better solution than making it very clear through docs (as I suggest above we can also add @storybook/addon-notes for extra notes in Storybook)

Reply:
> But if we don't define any of them, none of them will have any validation.

You mean propTypes validation? But do we really need to manage it? As far as we don't add any extra code around `onChange` callback, we don't need to validate it I guess. Or I don't get your point

Reply:
Yes I meant propTypes validation.

For the record, we decided not to explicitly define props like `onChange` or `onClick` but just propagate all of these through `{...rest}`. Will update `<Button />` as well.